### DISCUSSION TITLE
After transitioning through a transient state, best way to access the original event.type?

### DISCUSSION THREAD

**Discussion by awreccan** [statelyai/xstate#2128] [Q&A] [answered]
In my machine, I have a path through a transient state as follows:

`BACK` event -> transient state (choice pseudostate) -> `always`/`''` event -> one of many final states

These final states need to check `if (event.type === BACK)`
but `event.type` is `''` because the last event was `always`/`''`

One idea I have is to set a flag in the context.
Is there a different/better way? Is it possible to attach data to a `''` event, like `{ type: '', isBack: true }`

**Comment by ChrisShank** [comment, marked as answer]
I think this was recently answered [here](https://github.com/davidkpiano/xstate/discussions/2114#discussioncomment-625729).

**Reply by awreccan** [reply]
Thanks!
